src/ln/channelmanager.rs

```diff
@@ -2102,10 +2120,32 @@ impl ChainListener for ChannelManager {
 				}
 				true
 			});
+
 		}
 		for failure in failed_channels.drain(..) {
 			self.finish_force_close_channel(failure);
 		}
+
+		{
+			let mut channel_state = Some(self.channel_state.lock().unwrap());
+			for tx in txn_matched {
+				if let Some(payments_data) = self.monitor.is_resolving_output(&tx) {
```

Comment:
perhaps return type of is_resolving_output can be just vec! . if None implies vec.len() == 0

Reply:
btw awesome work, ariard! was studying carefully as I'm not familiar with on-chain transaction related code.

Reply:
Hmm started to just return vec!, but not sure, feel to me that current is more rustacean way of doing things!

Reply:
Thanks! In fact it's easier to understand I think if you have both BOLT2 and BOLT5 well in head